api.md

```diff
@@ -950,6 +1000,53 @@ SELECT device_id, time_bucket('5 minutes', time) as interval,
  alternative to an `ORDER BY time DESC LIMIT 1` clause to find the
  latest value (which will use indexes).
 
+## locf() [](locf)
+
+The `locf` function allows you to carry the last seen value in an aggregation group forward.
+It can only be used in an aggregation query with `time_bucket_gapfill`.
+The `locf` function call cannot be nested inside other function calls.
+
+#### Required Arguments [](locf-required-arguments)
+
+|Name|Description|
+|---|---|
+| `value` | The value to carry forward (anyelement) |
+
+#### Optional Arguments [](locf-optional-arguments)
+
+|Name|Description|
+|---|---|
+| `prev` | The lookup expression for values outside of the gapfill time range (anyelement) |
+
+#### Sample Usage [](locf-examples)
+
+Get the temperature every 5 minutes for each device over the 2 weeks carrying forward the last value for missing readings:
+```sql
+SELECT
+  time_bucket_gapfill('5 minutes', time, now() - interval '2 week', now()) as time,
+  device_id,
+  locf(avg(temperature))
+FROM metrics
+  WHERE time > now () - interval '2 week'
+GROUP BY 1,2
+ORDER BY 1 DESC;
+```
+
+Get the temperature every 5 minutes for each device over the 2 weeks carrying forward the last value for missing readings with out of bounds lookup
```

Comment:
See the above comment on interpolate about explaining need for "out-of-bounds" lookup (nit: note hyphenation), as well as example output.

Side note:  It seems like we really want a tutorial here showing timebucket alone, timebucket gapfill, timebucket gapfill + locf.   Should this be organized better in the reading-data section (as a contiguous mini-section), then these API docs could point there?  cc: @solugebefola

Reply:
done